docs/src/ref/mcmc.md

```diff
@@ -201,6 +201,207 @@ new_trace = permute_move(trace, 2)
+    m = sqrt(m1 * m2)
+    dof = m1 / (m1 + m2)
+    (m, dof)
+end
+```
+The inverse function is:
+```julia
+function split_mean(m, dof)
+    m1 = m * sqrt((dof / (1 - dof)))
+    m2 = m * sqrt(((1 - dof) / dof))
+    (m1, m2)
+end
+```
+We use these two functions to construct an involution, and we use this involution with [`metropolis_hastings`](@ref) to construct an MCMC kernel that we call a 'split/merge' kernel, because it either splits a parameter value, or merges two parameter values.
+The proposal is responsible for generating the extra degree of freedom when splitting:
+```julia
+@gen function split_merge_proposal(trace)
+    if trace[:z]
+        # currently two segments, switch to one
+    else
+        # currently one segment, switch to two
+        {:dof} ~ uniform_continuous(0, 1)
+    end
+end
+```
+Finally, we write the involution itself, using the involution DSL:
+```julia
+@involution function split_merge_involution(model_args, proposal_args, proposal_retval)
+
+    if @read_discrete_from_model(:z)
+
+        # currently two segments, switch to one
+        @write_discrete_to_model(:z, false)
+        m1 = @read_continuous_from_model(:m1)
+        m2 = @read_continuous_from_model(:m2)
+        (m, dof) = merge_means(m1, m2)
+        @write_continuous_to_model(:m, m)
+        @write_continuous_to_proposal(:dof, dof)
+
+    else
+
+        # currently one segments, switch to two
+        @write_discrete_to_model(:z, true)
+        m = @read_continuous_from_model(:m)
+        dof = @read_continuous_from_proposal(:dof)
+        (m1, m2) = split_mean(m, dof)
+        @write_continuous_to_model(:m1, m1)
+        @write_continuous_to_model(:m2, m2)
+    end
+end
+```
+The body of this function reads values from ``(t, u)`` at specific addresses and writes values to ``(t', u')`` at specific addresses, where ``t`` and ``t'`` are called 'model' choice maps, and ``u`` and ``u'`` are called 'proposal' choice maps.
+Note that the inputs and outputs of this function are **not** represented in the same way as arguments or return values of regular Julia functions --- they are implicit and can only be read from and written to, respectively, using a set of special macros (listed below).
+You should convince yourself that this function is invertible and its own inverse.
+
+Finally, we compose a structure-changing MCMC kernel using this involution:
+```julia
+split_merge_kernel(trace) = mh(trace, split_merge_proposal, (), split_merge_involution)
+```
+We then compose this move with the fixed structure move, and run it on the observed data:
+```julia
+(y1, y2) = (1.0, 1.3)
+trace, = generate(model, (), choicemap((:y1, y1), (:y2, y2), (:z, false), (:m, 1.)))
+for iter=1:100
+    trace = split_merge_kernel(trace)
+    trace = fixed_structure_kernel(trace)
+end
+```
+We can then compare the results to the results from the Markov chain that used the selection-based structure-changing kernel:
+
+![rjmcmc plot](images/rjmcmc.png)
```

Comment:
I think this image is not checked in?

Reply:
Fixed.

Reply:
Great.  Small suggestions on presentation for the image (GitHub doesn't allow commenting on the binary file):
* On the bottom plots, vertical axis, change the ticks from 0 and 1 to F and T.
* On the bottom plots, horizontal axis, add a label like "# inference steps".

Reply:
Done